### DISCUSSION TITLE
using ascom sky simulator but images not triggering when model building

### DISCUSSION THREAD

**Discussion by nigeldun** [mworion/MountWizzard4#355] [Q&A]
Hi

Looking for some help please…

I’m using the latest beta of MW4 with the ascom sky simulator and a 10micron mount simulator ( wip) and while I can trigger an image and solve it plus make the mount slew when I go to build a model it slews to the first position (with and without a dome) and queues the image but then nothing…. suspect its waiting for some condition to occur but not sure what. The mount is reporting that it’s tracking and the coords seem to all match so i’m at a bit of a loss. Do you have anything like a state diagram for what mw4 is expecting?

Also… any explanation as to how the satellite screens are supposed to work? I can get tles passing back and forth and calculate paths but nothing is appearing in the 2nd screen.

Anyone got any quick videos/screenshots of mw4 working as that would help me immensely.

Thanks for any help.

ps yes I know I’ve bitten off a bit of a large task :)

**Comment by mworion** [comment]
Hi,
It is a little bit hard to answer. So far I did not make a 10micron mount simulator as for me this was to complex despite the fact it would be really helpful in a virtual environment. So far I test MW4 functions with a real mount connected. MW4 needs some states through the mount protocol in the right order to derive states.
For the satellites: Please choose a satellite with a double click to show its paths in the second screen. Or if you already had selected a satellite and the mount is connected, then MW4 will retrieve its data automatically and shows it in the second screen.
Hope this helps.
Michel

If you are willing to finish a mount simulator, I will be happy using it for development work...

**Comment by nigeldun** [comment]
Hi Michel,
     hmm. Im at a bit of a loss as to what MW4 requires to trigger it to actually do the exposure for the model build. The exposure is lined up, the camera state is 0 (idle), the mount has slewed to the correct position and is in state tracking but it just sits there until I either cancel or end the build. It doesn’t seem to send any commands at this point other than those slow/medium/fast sets of commands. I’ve checked and they seem correct but I will check again. I did alter the mount state reported on the fly at one point and that did seem to trigger something to start happening so it looks like I’ve got something wrong there.

     The satellite tracking is a bit strange again as at one point I did get some details on the second screen but not been able to repeat that and again I’m not seeing any commands coming in.

     I don’t suppose you could possibly send me a log at trace level of when you trigger a model build and a satellite select at all. I appreciate that if you using a real mount you won’t want to mess things up but I can probably work out what’s going wrong from the logs.

     Unfortunately this is the “easy” bit, just wait until I implement the modelling:)

  Regards

     Nigel

**Comment by mworion** [comment]
Nigel,

Please post the actual log then I could check. It should be no problem to find out with the log.

Michel

**Comment by nigeldun** [comment]
Here you go. Started, switched to the right port, generated points then pressed run, waited and then stopped the model and shutdown.
[mw4-2024-04-29-MODELBUILD.log](https://github.com/mworion/MountWizzard4/files/15146203/mw4-2024-04-29-MODELBUILD.log)

**Comment by nigeldun** [comment]
Ok I’ve worked out what was going wrong with the satellite section when hooked up to my code. Seems the tle comes into my “mount” from mw4 with the new lines escaped as $0a while when it’s sent back mw4 expects the escape string to be $0A.As I was following the incoming format this caused things to not get processed inside mw4.

There also seems to be some flexibility in the format of julian date strings which vary from 8 to 10 decimal places and the position coords which have up to 15 decimal places in the satellite commands such as TRADD and TLEP etc. As these commands don’t appear in any version of the documentation I can find on the web this could be as expected and it’s easy enough to handle but just seems inconsistent with the other commands which have a very specific format.

**Comment by mworion** [comment]
I fully agree, the documentation is not 100% especially when string length etc. is concerned. I would be happy to step in and use the simulator for test purposes.Do you have any information, how you are building it (github etc).Am 30.04.2024 um 14:08 schrieb nigeldun ***@***.***>:﻿
Ok I’ve worked out what was going wrong with the satellite section when hooked up to my code. Seems the tle comes into my “mount” from mw4 with the new lines escaped as $0a while when it’s sent back mw4 expects the escape string to be $0A.As I was following the incoming format this caused things to not get processed inside mw4.
There also seems to be some flexibility in the format of julian date strings which vary from 8 to 10 decimal places and the position coords which have up to 15 decimal places in the satellite commands such as TRADD and TLEP etc. As these commands don’t appear in any version of the documentation I can find on the web this could be as expected and it’s easy enough to handle but just seems inconsistent with the other commands which have a very specific format.

**Comment by nigeldun** [comment]
I’ve now worked out why the model building run wasn’t working. The :MS# documentation says to slew and then track, so my internal code says slewing and tracking at the same time which means MW4 ends up being told the mount is tracking all the time when it’s expecting slewing then tracking.
Looking at the ascom docs they seem to say tracking must be true when doing a slew to coordinates, though it doesn’t really make sense to me that you can be in both states at the same time. I will have to have a think.

My code is in github but in a private repo (mainly to save embarrassment). It’s going under quite rapid development at the moment as I'm trying to get to the point of actually building models to see if I can. When I get to that point I will probably open it up or scrap it if I fail miserably.
